src/ParallelAlgorithms/Events/ObserveFields.hpp

```diff
@@ -204,7 +204,52 @@ class ObserveFields<VolumeDim, ObservationValueTag, tmpl::list<Tensors...>,
       const OptionalAnalyticSolutions& optional_analytic_solutions,
       Parallel::GlobalCache<Metavariables>& cache,
       const ElementId<VolumeDim>& array_index,
-      const ParallelComponent* const /*meta*/) const noexcept {
+      const ParallelComponent* const component) const noexcept {
+    call_operator_impl(subfile_path_, variables_to_observe_,
+                       interpolation_mesh_, observation_value, mesh,
+                       inertial_coordinates, analytic_solution_tensors...,
+                       non_solution_tensors..., optional_analytic_solutions,
+                       cache, array_index, component);
+  }
+
+  // This overload is called when the list of analytic-solution tensors is
+  // empty, i.e. it is clear at compile-time that no analytic solutions are
+  // available
+  template <typename Metavariables, typename ParallelComponent>
+  void operator()(
+      const typename ObservationValueTag::type& observation_value,
+      const Mesh<VolumeDim>& mesh,
+      const tnsr::I<DataVector, VolumeDim, Frame::Inertial>&
+          inertial_coordinates,
+      const typename NonSolutionTensors::type&... non_solution_tensors,
+      Parallel::GlobalCache<Metavariables>& cache,
+      const ElementId<VolumeDim>& array_index,
+      const ParallelComponent* const component) const noexcept {
+    this->operator()(observation_value, mesh, inertial_coordinates,
+                     non_solution_tensors..., std::nullopt, cache, array_index,
+                     component);
+  }
+
+  // We factor out the work into a static member function so it can  be shared
+  // with other field observing events, like the one that deals with DG-subcell
+  // where there are two grids. This is to avoid copy-pasting all of the code.
+  template <typename OptionalAnalyticSolutions, typename Metavariables,
+            typename ParallelComponent>
+  static void call_operator_impl(
```

Comment:
Why make this static and pass in the member variables, instead of just non-static and using the member variables like before?

Reply:
I don't want to have to have an object of `dg::Events::ObserveFields` inside `evolution::dg::subcell::Events::ObserveFields` is the big reason. I would've moved this implementation into a `detail` namespace but since the function signature uses the template parameters (specifically, the expanded packs), this was easier :shrug: